frontend/dashboard/src/app/page.tsx

```diff
@@ -1,111 +1,200 @@
-import Image from "next/image";
+"use client";
+import { Button } from "@/components/ui/button";
+import {InputWithLabel} from "@/components/ui/inputWithLabel";
+import { SelectScrollable } from "@/components/ui/selectScrollable";
+import { College, Interest } from "@/const/dummyConst";
+import MultipleSelector from '@/components/ui/multiSelect';
+import { useForm, Controller } from "react-hook-form";
+import { z } from "zod";
+import {Option} from "@/types/option";
+import { useState } from 'react';
+import { zodResolver } from "@hookform/resolvers/zod";
+
+type FormData = {
+  firstName: string;
+  lastName: string;
+  email: string;
+  college: string;
+  interests: Option[];
+};
+
+const formSchema = z.object({
```

Comment:
speaking on using the lib based schemas, we can still use the validation we were using in the controllers in this by using refine:

const formSchema = z.object({
  firstName: z.string()
    .min(2, "First name must be at least 2 characters")
    .refine(value => /^[a-zA-Z]+$/.test(value), {
      message: "First name can only contain alphabetic characters",
    }),
  lastName: z.string()
    .min(2, "Last name must be at least 2 characters")
    .refine(value => /^[a-zA-Z]+$/.test(value), {
      message: "Last name can only contain alphabetic characters",
    }),
  email: z.string()
    .email("Please enter a valid email")
    .refine(value => /^[a-zA-Z]+\.[a-zA-Z]@northeastern\.edu$/.test(value), {
      message: "Please enter your Northeastern email",
    }),
  college: collegeEnum.refine(value => value !== undefined, {
    message: "Please select a college",
  }),
  interests: z.array(z.object({
    label: z.string(),
    value: z.string(),
  })).nonempty("At least one interest is required"),
});